### PR TITLE
Sort init import

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

Not a high-priority item but I get bored at nights and I like writing those kinds of scripts 😅

So this PR adds a script to properly sort the import inside `_import_structure` because people have been absolutely ruthless and putting their objects in any kind of random order. That's not very feng-shui so I'm bringing back harmony by having the same sort as isort applied to all `__init__` that contain an `_import_structure`.

### CODE REVIEW

File: src/transformers/__init__.py [hunk 1]
```diff
@@ -2170,7 +2166,7 @@
             TFLxmertPreTrainedModel,
             TFLxmertVisualFeatureEncoder,
         )
-        from .models.marian import TFMarian, TFMarianMTModel
+        from .models.marian import TFMarianModel, TFMarianMTModel
```
Comment (sgugger) [inline src/transformers/__init__.py:2169]: Completely unrelated to this PR, renamed that object to its proper name.

File: Makefile [hunk 2]
```diff
@@ -26,22 +26,23 @@ extra_quality_checks: deps_table_update
 	python utils/check_table.py
 	python utils/check_dummies.py
 	python utils/check_repo.py
-	python utils/style_doc.py src/transformers docs/source --max_len 119
+	python utils/style_doc.py src/transformers docs/source --max_len 119 --check_only
```
Comment (LysandreJik) [inline Makefile:29]: If I'm not mistaken, this prevents your `style_doc.py` to do changes when used with `make fixup`, is that voluntary?

File: Makefile [hunk 3]
```diff
@@ -26,22 +26,23 @@ extra_quality_checks: deps_table_update
...
 	python utils/class_mapping_update.py
 
 # this target runs checks on all files
 quality:
 	black --check $(check_dirs)
 	isort --check-only $(check_dirs)
+	python utils/custom_init_isort.py --check_only
```
Comment (LysandreJik) [inline Makefile:36]: I would put this in the `extra_quality_checks` too so that it runs with `make fixup`

File: utils/custom_init_isort.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Inc. team.
```
Comment (LysandreJik) [inline utils/custom_init_isort.py:2]:
```suggestion
# Copyright 2021 The HuggingFace Inc. team.
```

**Review by LysandreJik — APPROVED** [review]
Very nice! LGTM

**Comment by sgugger** [comment]
To address your comments on the `MakeFile`, I have removed some checks from `extra_quality_checks` because they are checks that modify content and `make quality` is only supposed to check, not change.

To have `make fixup` still work as intended, I put the checks that change content in `extra_style_checks` that is called both by `make fixup` and `make style`. Could you double-check it looks okay @LysandreJik and @stas00 ? Thanks!

**Review by LysandreJik — APPROVED** [review]
Your changes LGTM. I agree with you that @stas00's review would be welcome before we merge.

File: Makefile [hunk 5]
```diff
@@ -21,32 +21,36 @@ deps_table_update:
...
 # Check that source code meets quality standards
 
-extra_quality_checks: deps_table_update
+extra_quality_checks:
 	python utils/check_copies.py
 	python utils/check_table.py
 	python utils/check_dummies.py
 	python utils/check_repo.py
-	python utils/style_doc.py src/transformers docs/source --max_len 119
 	python utils/class_mapping_update.py
```
Comment (stas00) [inline Makefile:29]: should we create a new target for auto-generation? It'd be:
```
autogenerate_code: deps_table_update
        python utils/class_mapping_update.py
```

**Review by stas00 — APPROVED** [review]
LGTM

Added another small tweak suggestion

File: Makefile (hunk 5 above)
Comment (sgugger) [reply Makefile:29]: Oh I hadn't realized `class_mapping_update` was writing things. It should also disappear from `make quality` then to go with the others in `extra_style_checks`.

For me:

`make fixup` -> applies the styling patches and checks everything passes quickly
`make style`  -> applying styling patches and other auto-generated code
`make quality` -> only checks, does not write anything

Comment (stas00) [reply Makefile:29]: I only ever use `make fixup` - the initial design was that `fixup` = `style+quality` on only modified files.

So it was created before we added 2 items that generated code, so if others prefer the quick version then `fixup` should include the autogenerate target too.

Actually, is there is a situation where `fixup` won't do the right thing already? When does one need to run the full `make style`?

Comment (stas00) [reply Makefile:29]: class_mapping_update doesn't do anything if the relevant file hasn't changed so it's blazingly fast
we can do the same for deps_table_update so it'd be instant 99% of the time.

Comment (sgugger) [reply Makefile:29]: Yes `fixup` is `style+quality` on only modified files. My problem is that some processes that modify files have been added to `quality-extra-checks` which is run by `quality` on top of `fixup`. So `fixup` runs `extra-quality-checks` as well as `extra-styling-checks` to do both, I don't want to change that part.

Comment (sgugger) [reply Makefile]: See latest version for my proposal:
- `extra-quality-checks` common processes for quality and fixup, checks that are fast or need to run on all files, not changing any file
- `extra-style-checks` common processes for style and fixup, checks that are fast or need to run on all files, changing files if necessary
- `style` style all files + other checks that are "writing" the right thing
- `quality` check all files + other checks (in check only mode)
-  `fixup` same as `style` + `quality` but only on modified files.

File: Makefile [hunk 6]
```diff
@@ -21,32 +21,36 @@ deps_table_update:
...
 # Format source code automatically and check is there are any problems left that need manual fixing
 
-style: deps_table_update
+extra_style_checks: deps_table_update
+	python utils/custom_init_isort.py
+	python utils/style_doc.py src/transformers docs/source --max_len 119
+	python utils/class_mapping_update.py
+
+# this target runs checks on all files
+style:
```
Comment (stas00) [inline Makefile:46]: this could just be:
```
style: extra_style_checks
```

for some reason it doesn't let me propose the edit across several lines

Comment (stas00) [reply Makefile:46]: unless you need that target to run after black and isort then it's perfect as it is.

Comment (sgugger) [reply Makefile:46]: I am not 100% sure the scripts I wrote will work on badly formatted data, so if it's all the same to you, I would leave it at the end just to be sure it works.

Comment (stas00) [reply Makefile:46]: yes, yes, it's a perfectly good way

**Pull request merged**